.github/workflows/rust_checks.yml

```diff
@@ -0,0 +1,128 @@
+name: Rust Static checks
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/policy_controller.yml
+      - Cargo.lock
+      - Cargo.toml
+      - charts/linkerd-control-plane/templates/destination-rbac.yaml
+      - charts/linkerd-crds/templates/policy/**
+      - deny.toml
+      - policy-controller/**
+      - policy-test/**
+      - rust-toolchain
+
+permissions:
+  contents: read
+
+env:
+  CARGO_INCREMENTAL: 0
+  CARGO_NET_RETRY: 10
+  K3D_VERSION: v5.4.1
+  PROTOC_NO_VENDOR: 1
+  PROXY_INIT_VERSION: v1.5.3
+  RUST_BACKTRACE: short
+  RUSTUP_MAX_RETRIES: 10
+
+
+jobs:
+  fmt:
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
+    container:
+      image: docker://rust:1.60.0
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+      - uses: olix0r/cargo-action-fmt@ee1ef42932e44794821dab57ef1bf7a73df8b21f
+      - run: rustup component add rustfmt
+      - run: cargo fmt --all -- --check | cargo-action-fmt
+
+  audit:
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        checks:
+          - advisories
+          - bans licenses sources
+    # Prevent sudden announcement of a new advisory from failing Ci.
+    continue-on-error: ${{ matrix.checks == 'advisories' }}
+    steps:
+    - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+    - uses: EmbarkStudios/cargo-deny-action@30ecad1d5873c1cc2cad10a33637371ca094768b
+      with:
+        command: check ${{ matrix.checks }}
+
+  clippy:
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    container:
+      image: docker://rust:1.60.0
+    steps:
+      - run: rustup component add clippy
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+      - uses: olix0r/cargo-action-fmt@ee1ef42932e44794821dab57ef1bf7a73df8b21f
+      - run: cargo fetch --locked
+      - name: Run cargo clippy
+        run: cargo clippy --frozen --all --no-deps --message-format=json | cargo-action-fmt
```

Comment:
You didn't change this, but we should run clippy on all targets (i.e. tests as well as app code). `--all` has been deprecated in favor of `--workspace`

```suggestion
        run: cargo clippy --frozen --workspace --all-targets --message-format=json | cargo-action-fmt
```

Reply:
Ah, my bad, I have just checked the [cargo check](https://doc.rust-lang.org/cargo/commands/cargo-check.html) reference, wasn't aware that by default the tests aren't checked. TIL.